File: RoombaServerStreaming/servidorStreaming.rb
require 'socket'
require 'Pathname'
#Para hacer que esto funcione debe haber en tu home una carpeta llamada LibreriaMusica y adentro 
#/Yeah Yeah Yeahs/Fever/maps.mp3 (u otra canción, pero deberá cambiarse el nombre del archivo en el cliente)
class ServidorStreaming
    #Aquí utilizo un buffer de 4096
    $BUFFER_SIZE = 4096
    DIRECCION = "127.0.0.1"
    PUERTO = 1234
    $servidor = UDPSocket.new
    $servidor.bind(DIRECCION, PUERTO)
    puts "Servidor en espera de clientes"
    def ServidorStreaming.recuperarCancion(ruta)
        rutaInicio = Dir.home
        rutaInicio = rutaInicio + "/LibreriaMusica/"
        pathRuta = Pathname(rutaInicio + ruta)
        puts pathRuta
        if pathRuta.exist?
            puts "La encontró"
            archivo = File.open(pathRuta, "r")
            return archivo
        end
    end
    def self.enviarCancion(archivo, remitente)
        while chunk = archivo.read($BUFFER_SIZE)
            $servidor.send(chunk, 0, remitente[2], remitente[1])
        end
        archivo.close
    end

    loop do
        puts "inicio"
        ruta, remitente = $servidor.recvfrom($BUFFER_SIZE)
        puts "Cliente conectado"
        archivo = ServidorStreaming.recuperarCancion(ruta)
        puts "Size #{archivo.size}"
        ServidorStreaming.enviarCancion(archivo, remitente)

    end
end

File: RoombaServerStreaming/ServidorStreaming.py
import socket
import os
from os.path import expanduser
import threading

class ServidorThread():
    def __init__(self):
        self.host = ''
        self.port = 1235
        self.serverSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.serverSocket.bind((self.host, self.port))    

    def escucha (self):
        self.serverSocket.listen(15)
        while True:
            print "Servidor en espera de clientes"
            cliente, address = self.serverSocket.accept()
            threading.Thread(target = self.recuperarArchivo, args = (cliente, address)).start()
    @staticmethod
    def enviarDatos(archivo, cliente):
        BUFFER_SIZE = 4096
        print "Envia datos al cliente"
        try:
            chunk = archivo.read(BUFFER_SIZE)
            while chunk != "":
                cliente.send(chunk)
                chunk = archivo.read(BUFFER_SIZE)
        finally:
            cliente.close()
            archivo.close()
        return

    def recuperarArchivo(self, cliente, address):
        BUFFER_SIZE = 4096
        #home = expanduser("~")
        home = os.getcwd() + "/LibreriaMusica/"

        ruta = cliente.recv(BUFFER_SIZE)
        pathArchivo = home + ruta
        print pathArchivo
        if os.path.isfile(pathArchivo):
            print "Recupero archivo"
            archivo = open(pathArchivo, "rb")

            self.enviarDatos(archivo, cliente)  
                    
if __name__ == '__main__':  
    ServidorThread().escucha()